crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -38,15 +40,16 @@ use tracing::trace;
 const MAX_HEADERS_RANGE: u64 = 1_000; // with ~530bytes per header this is ~500kb
 
 /// `Eth` filter RPC implementation.
-pub struct EthFilter<Provider, Pool> {
+pub struct EthFilter<Provider, Pool, Events> {
     /// All nested fields bundled together
-    inner: Arc<EthFilterInner<Provider, Pool>>,
+    inner: Arc<EthFilterInner<Provider, Pool, Events>>,
```

Comment:
hmm, not sure we need the additional generic here.
because this is never used internally, we can get rid of this if we spawn the watch task externally